app/src/main/java/com/codecool/lolapp/viewmodels/ListingViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.codecool.lolapp.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.codecool.lolapp.model.Character
+import com.codecool.lolapp.model.CharactersApi
+import com.codecool.lolapp.model.Response
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableSingleObserver
+import io.reactivex.schedulers.Schedulers
+
+class ListingViewModel(private var charactersApi: CharactersApi) : ViewModel() {
+
+    private var disposable = CompositeDisposable()
+    val characters = MutableLiveData<List<Character>>()
```

Comment:
I don't think you need 3 separate channels for this. Try creating a common parent for the response data which has these 3 different states as child classes, so you can send the result on the same LiveData